src/python/gudhi/representations/vector_methods.py

```diff
@@ -427,26 +401,65 @@ def fit_transform(self, X, y = None):
             starts = offsets[0:N]
             ends = offsets[1:N + 1] - 1
 
-            xs = [-np.inf]
             bettis = [[0] for i in range(0, N)]
 
-            for i in sorting:
-                j = np.searchsorted(ends, i)
-                delta = 1 if i - starts[j] < len(X[j]) else -1
-                if events[i] == xs[-1]:
+            i = 0
+            for x in self.grid_:
+                while i < len(sorting) and events[sorting[i]] <= x:
+                    j = np.searchsorted(ends, sorting[i])
+                    delta = 1 if sorting[i] - starts[j] < len(X[j]) else -1
                     bettis[j][-1] += delta
-                else:
-                    xs.append(events[i])
-                    for k in range(0, j):
-                        bettis[k].append(bettis[k][-1])
-                    bettis[j].append(bettis[j][-1] + delta)
-                    for k in range(j+1, N):
-                        bettis[k].append(bettis[k][-1])
+                    i += 1
+                for k in range(0, N):
+                    bettis[k].append(bettis[k][-1])
+    
+            return np.array(bettis, dtype=int)[:, 0:-1]
 
-            self.grid_ = np.array(xs)
-            return np.array(bettis, dtype=int)
+    def fit_transform(self, X):
```

Comment:
Wait what? Is this reverting #1061? Bad merge?

Reply:
weird, I must have messed up with the merge, fixed it

Reply:
Hmm, now you seem to have a merge issue with the gudhi-deploy submodule.
(yes, git submodules are a pain)